cmd/analyzer/main.go

```diff
@@ -156,39 +120,23 @@ func diffAnalysisMain(inArgs *InArgs, vpcConfigs map[string]*vpcmodel.VPCConfig,
 	if err1 != nil {
 		return fmt.Errorf(ErrorFormat, ParsingErr, err1)
 	}
-	vpc2ndConfigs, err4 := ibmvpc.VPCConfigsFromResources(rc2ndForDiff, *inArgs.VPC, *inArgs.Debug)
-	if err4 != nil {
-		return fmt.Errorf(ErrorFormat, InGenerationErr, err4)
+	vpc2ndConfigs, err2 := ibmvpc.VPCConfigsFromResources(rc2ndForDiff, *inArgs.VPC, *inArgs.Debug)
```

Comment:
So now the only difference between the general case and the "diff" case is that in the latter we should make sure that there is only one `VPC` given. It seems that a separate function `diffAnalysisMain` is no longer needed; all that is required is an extra check for diff analysis (that there is at most one `VPC`). Why not have this check in the parser, and avoid the separation here?

Reply:
in diffAnalysisMain() we also calls ParseResourcesFromFile() and VPCConfigsFromResources() for the second config

Reply:
Wouldn't it be better to add the check I suggested to the parser and then call the above functions in case the 2nd file is specified? Parser already verifies that if a 2nd file is specified then the mode is a diff one
Apart from avoiding code duplication this solution seems more general (e.g. in case there will be another mode with 2 files).

Reply:
fixed